libcalico-go/test/mock-policies.yaml

```diff
@@ -0,0 +1,62 @@
+#These policies are used for testing if the correct policy name is returned when no name is part of the projectcalico.org/metadata annotation
```

Comment:
I'd really like to avoid checking in a YAML file as part of the tests - we should be aiming for UTs to not rely on external configuration ( I know this isn't already the case, but let's not make it any worse! )

We don't need to block this PR on the test - we can submit a follow-on to add it. I have ideas on how to do this without needing externally applied policies.